__tests__/app.test.js

```diff
@@ -414,3 +414,49 @@ describe("10. GET /api/reviews/:review_id (comment count)", () => {
       });
   });
 });
+
+describe("11. GET /api/reviews (queries)", () => {
+  test("GET - 200 returns views object for the passed category.", () => {
+    return request(app)
+      .get("/api/reviews?category='social deduction'")
+      .expect(200)
+      .then(({ body }) => {
+        body.reviews.forEach((review) => {
+          expect(review).toMatchObject({
+            owner: expect.any(String),
+            title: expect.any(String),
+            category: "social deduction",
+            review_img_url: expect.any(String),
+            review_id: expect.any(Number),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            designer: expect.any(String),
+          });
+        });
+      });
+  });
+  test("GET - 200 returns views object sorted by the passed sort_by value", () => {
+    return request(app)
+      .get("/api/reviews?sort_by=designer")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.reviews[0].designer).toBe("Wolfgang Warsch");
+      });
+  });
+  test("GET - 200 returns views object sorted in ascending or descending order based on the order value passed", () => {
+    return request(app)
+      .get("/api/reviews?sort_by=designer&order=ASC")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.reviews[0].designer).toBe("Akihisa Okui");
+      });
+  });
+  test("GET - 400 returns an error invalid sort query when passed invalid query value", () => {
+    return request(app)
+      .get("/api/reviews?sort_by=nonsense")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid sort query!");
+      });
+  });
```

Comment:
Need a couple more tests here for error handling. You're checking for invalid sort_by's but what about invalid categorys, or orders?

Reply:
sorry I did handle the invalid order in my model but forgot to do the test for it, with the category I thought we don't need it because if they send an invalid category then it will return empty object which is not considered as an error. but anyway I will try to do the test for that